src/e2e/java/uk/gov/hmcts/reform/sscs/functional/handlers/AdminAppealWithdrawnNotificationsTest.java

```diff
@@ -0,0 +1,118 @@
+import static uk.gov.hmcts.reform.sscs.domain.notify.NotificationEventType.CASE_UPDATED;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import junitparams.Parameters;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Correspondence;
+import uk.gov.hmcts.reform.sscs.ccd.domain.CorrespondenceType;
+import uk.gov.hmcts.reform.sscs.functional.AbstractFunctionalTest;
+import uk.gov.hmcts.reform.sscs.functional.Retry;
+import uk.gov.service.notify.Notification;
+
+
+public class AdminAppealWithdrawnNotificationsTest extends AbstractFunctionalTest {
+
+    public AdminAppealWithdrawnNotificationsTest() {
+        super(30);
+    }
+
+    @Rule
+    public Retry retry = new Retry(0);
+
+    @Rule
+    public Timeout globalTimeout = Timeout.seconds(120);
+
+    @Before
+    public void setUp() {
+        delayInSeconds(10); //Allow some time to finish potential callbacks as part of the AppealCreated event
+        initialiseCcdCase();
+    }
+
+    @Test
+    @Parameters({
+        "Appellant, 8620e023-f663-477e-a771-9cfad50ee30f, 446c7b23-7342-42e1-adff-b4c367e951cb, 1, 1, 1",
+        "Appointee, 8620e023-f663-477e-a771-9cfad50ee30f, 446c7b23-7342-42e1-adff-b4c367e951cb, 1, 1, 1",
+        "Reps, e29a2275-553f-4e70-97f4-2994c095f281, f59440ee-19ca-4d47-a702-13e9cecaccbd, 1, 1, 2"
+    })
+    public void givenCallbackWithSubscriptions_shouldSendEmailSmsAndLetterNotifications(
+        String subscription,
+        String emailId,
+        String smsId,
+        int expectedNumEmailNotifications,
+        int expectedNumSmsNotifications,
+        int expectedNumLetters) throws Exception {
+
+        simulateCcdCallback(ADMIN_APPEAL_WITHDRAWN, "handlers/" + ADMIN_APPEAL_WITHDRAWN.getId() + subscription
+            + "Callback.json");
+
+        List<Notification> notifications = tryFetchNotificationsForTestCase(emailId, smsId);
+
+        assertEquals(expectedNumEmailNotifications, getNumberOfNotificationsForGivenEmailOrSmsTemplateId(notifications, emailId));
+        assertEquals(expectedNumSmsNotifications, getNumberOfNotificationsForGivenEmailOrSmsTemplateId(notifications, smsId));
+        assertTrue(fetchLetters(expectedNumLetters));
+    }
+
+    private boolean fetchLetters(int expectedNumLetters) {
+        do {
+            if (getNumberOfLetterCorrespondence() == expectedNumLetters) {
+                return true;
+            }
+            delayInSeconds(5);
+        } while (true);
+    }
+
+    @Test
+    public void givenCallbackWithNoSubscription_shouldSendLetterNotifications() throws Exception {
+        simulateCcdCallback(ADMIN_APPEAL_WITHDRAWN, "handlers/" + ADMIN_APPEAL_WITHDRAWN.getId()
+            + "NoSubscriptions" + "Callback.json");
+
+        String emailId = "8620e023-f663-477e-a771-9cfad50ee30f";
+        String smsId = "446c7b23-7342-42e1-adff-b4c367e951cb";
```

Comment:
Is there anyway some of these can be moved to integration tests? The functional tests for notifications take a long time. We were trying to implement a pattern of one notification template covered in one test, but this looks like the same template is covered in 2 tests? Any tests covering logic changes should be covered in integration tests.